### PR TITLE
Prevents user from logging out via cancelled /user requests

### PR DESCRIPTION
**Pull request by asg5704**
## Description
**Issue**: [https://github.com/department-of-veterans-affairs/va.gov-team/issues/13720](https://github.com/department-of-veterans-affairs/va.gov-team/issues/13720)

This PR adds more logic to the catch statement in the `initializeProfile` action.  The `catch` statement has a conditional statement to check if the error is both a TypeError & has a Failed to Fetch message.  - - When `truthy` an additional `GET /user` request is made to ensure we are actually receiving a logged out response from the endpoint.
- When `falsy` the profile sessions is torn down.
## Testing done
Manual / Visual testing

## Acceptance criteria
- [X] User is not logged out because of the `GET user/` endpoint was cancelled
- [X] Other `GET user/` call failures should log out users as normal

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by erikphansen — APPROVED** [review]
In my tests locally this fixes the issue; a well-time reload of the page while the `GET user/` call is pending no longer signs you out. But I wonder if the second call to `GET user` should even be done. Is there a scenario where getting a `Failed to fetch` `TypeError` should actually result in the user getting signed out?

File: src/platform/user/profile/actions/index.js [hunk 1]
```diff
@@ -46,8 +46,19 @@ export function initializeProfile() {
       await dispatch(refreshProfile());
       dispatch(updateLoggedInStatus(true));
     } catch (error) {
-      dispatch(updateLoggedInStatus(false));
-      teardownProfileSession();
+      /* If the fetch fails the first time, try again to ensure the profile should actually be torn down */
+      if (error instanceof TypeError && error.message === 'Failed to fetch') {
```
[the remark this answers is not in the document]
Comment (AGarcia-Clarity) [reply src/platform/user/profile/actions/index.js:50]: I did try to search as much as possible for the Error messages but had not such luck.  I could try to dig a little deeper and see if there is anything.

File: src/platform/user/profile/actions/index.js [hunk 2]
```diff
@@ -46,8 +46,19 @@ export function initializeProfile() {
       await dispatch(refreshProfile());
       dispatch(updateLoggedInStatus(true));
     } catch (error) {
-      dispatch(updateLoggedInStatus(false));
-      teardownProfileSession();
+      /* If the fetch fails the first time, try again to ensure the profile should actually be torn down */
```
[the remark this answers is not in the document]
Comment (AGarcia-Clarity) [reply src/platform/user/profile/actions/index.js:49]: I can update the language as appropriate.

**Comment by AGarcia-Clarity** [comment]
> In my tests locally this fixes the issue; a well-time reload of the page while the `GET user/` call is pending no longer signs you out. But I wonder if the second call to `GET user` should even be done. Is there a scenario where getting a `Failed to fetch` `TypeError` should actually result in the user getting signed out?

So I don't think the additional `GET /user` needs to be called however I just didn't want to do a short-circuit return statement (maybe this is the direction we want to go).  I am still trying to get the e2e tests to run on my computer and will updates with your feedback.  Thanks 😎

File: src/platform/user/profile/actions/index.js [hunk 3]
```diff
@@ -46,8 +46,14 @@ export function initializeProfile() {
       await dispatch(refreshProfile());
       dispatch(updateLoggedInStatus(true));
     } catch (error) {
-      dispatch(updateLoggedInStatus(false));
-      teardownProfileSession();
+      /* If the fetch fails due to the browser cancelling the request due to a navigation event short circuit it to prevent terminating session */
+      if (error instanceof TypeError && error.message === 'Failed to fetch') {
+        // eslint-disable-next-line no-useless-return
+        return;
```
Comment (bkjohnson) [inline src/platform/user/profile/actions/index.js:52]: Is this `return` necessary? It doesn't look like there is anything after the if statement that would happen if the `return` were removed.

**Review by bkjohnson — COMMENTED** [review]
I think it would be good if the if statement could be changed to something like:


```js
      if (!(error instanceof TypeError && error.message === 'Failed to fetch')) {
        dispatch(updateLoggedInStatus(false));
        teardownProfileSession();
      }
```

or 

```js
      if (!(error instanceof TypeError) || error.message !== 'Failed to fetch')) {
        dispatch(updateLoggedInStatus(false));
        teardownProfileSession();
      }
```

File: src/platform/user/profile/actions/index.js (hunk 3 above)
Comment (AGarcia-Clarity) [reply src/platform/user/profile/actions/index.js:52]: I'm going to refactor it to what you suggested and see if the same results.  Initially, any caught error tore down the profile but I will let you know what I find out with testing.

Comment (AGarcia-Clarity) [reply src/platform/user/profile/actions/index.js]: Looks like it works as expected without the return statement; Pushing up modifications now.  Thanks for your suggested refactor 👍

**Review by bkjohnson — APPROVED** [review]

**Pull request merged**